go-controller/pkg/ovn/egressfirewall_dns.go

```diff
@@ -200,10 +200,14 @@ func (e *EgressDNS) Run(defaultInterval time.Duration) {
 			// find the domain name whose DNS entry will expire first and calculate when it will expire,
 			// set timer to what's sooner: default update interval or next expiration time
 			ttl, domainNameExpiringNext, timeSet = e.dns.GetNextQueryTime()
-			if time.Until(ttl) > defaultInterval || !timeSet {
+			ttlDuration := time.Until(ttl)
+			if ttlDuration > defaultInterval || !timeSet {
 				durationTillNextQuery = defaultInterval
-			} else {
+			} else if ttlDuration.Seconds() > 0 {
 				durationTillNextQuery = time.Until(ttl)
+			} else {
```

Comment:
I think is better to set `durationTillNextQuery = ttlDuration`, since new return value from `time.Until(ttl)` is not guaranteed to be > 0

Reply:
That's right. good catch. changed it.